src/components/Inputs/Schemas.js

```diff
@@ -0,0 +1,135 @@
+import { Button } from '@dhis2/ui-core'
+import { connect } from 'react-redux'
+import { useForm } from 'react-final-form'
+import React, { useEffect, useCallback } from 'react'
+import i18n from '@dhis2/d2-i18n'
+import propTypes from 'prop-types'
+
+import { Checkbox } from '../FinalFormComponents/Checkbox'
+import { Loading } from '../Loading'
+import { fetchSchemas } from '../../reducers/schemas/thunks'
+import {
+    getGroupLables,
+    getGroupOrder,
+    getSchemaGroups,
+} from '../../reducers/schemas/selectors'
+import s from './Schemas.module.css'
+
+const SCHEMAS_KEY = 'schemas'
+
+const ToggleAll = ({ schemaKey, value, children, className }) => {
+    const form = useForm()
+
+    const onClick = useCallback(() => {
+        const fieldKeys = Object.keys(form.getState().values[schemaKey])
+
+        form.batch(() => {
+            fieldKeys.forEach(fieldKey => {
+                form.change(`${schemaKey}.${fieldKey}`, value)
+            })
+        })
+    }, [form, value, schemaKey])
+
+    return (
+        <Button
+            onClick={onClick}
+            type="button"
+            children={children}
+            className={className}
+        />
+    )
+}
+
+const SelectAll = ({ schemaKey }) => (
+    <ToggleAll schemaKey={schemaKey} value={true} className={s.selectAllButton}>
+        {i18n.t('Select All')}
+    </ToggleAll>
+)
+
+const SelectNone = ({ schemaKey }) => (
+    <ToggleAll schemaKey={schemaKey} value={false}>
+        {i18n.t('Select None')}
+    </ToggleAll>
+)
+
+const Controls = ({ schemaKey }) => (
+    <div className={s.controls}>
+        <SelectAll schemaKey={schemaKey} />
+        <SelectNone schemaKey={schemaKey} />
+    </div>
+)
+
+const SchemaGroup = ({ label, schemas }) => (
+    <div className={s.group}>
+        <span className={s.formLabel}>{label}</span>
+        <div className={s.schema}>
+            {schemas.map(schema => (
+                <div key={schema.name}>
+                    <Checkbox
+                        label={schema.label}
+                        name={`schemas.${schema.name}`}
+                        checkedInitially
+                    />
+                </div>
+            ))}
+        </div>
+    </div>
+)
+
+const SchemasInput = ({
+    excludeSchemas,
+
```

Comment:
empty line

Reply:
We should have a lint rule for this.

Reply:
I don't think we should have a rule for this.
Sometimes it's really handy to group props with spaces (especially when there are like 10+ args)

Reply:
Yeah I think our approaches differ in that regard. I always find the whitespace between props, imports, etc. a bit distracting, to me it hinders scanning. Sometimes I find it hard to tell what they were grouped by, as there's no explicit comment. But ok, not the most important thing in the world.